### DISCUSSION TITLE
Is there any way that we can save linker object in spark after training and predicting

### DISCUSSION THREAD

**Discussion by rapidAmbakar** [moj-analytical-services/splink#827] [Q&A] [answered]
I have just predicted on a sparkdata frame which take roughtly around 1.5 hours. Is the return object coming from linker.predict() is just limited to save it to dataframe? or is their any any which i dont have to predict every time.

**Comment by RobinL** [comment, marked as answer]
To save the outputs you will need to save linker.predict() to a dataframe using:
```
df_predict.as_pandas_dataframe()
```
or
```
df_predict.as_spark_dataframe().write.parquet("output_folder") 
```

To make it run faster, you could consider using tighter blocking rules, see 
https://moj-analytical-services.github.io/splink/demos/02_Blocking.html
